src/es6/SelflessEntity.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+/**
+ * Abstract Entity class to help create entity classes.
+ */
+export class SelflessEntity {
+	/**
+	 * Primes the object used by the entityFactory. Should never be called outside.
+	 * @param {Object} sdkParentEntity Parent entity that is a Entity.
+	 * @param {Object} entity A hypermedia siren entity as defined by [the siren specification]{@link https://github.com/kevinswiber/siren}
+	 */
+	constructor(sdkParentEntity, entity) {
+		if (this.constructor === SelflessEntity) {
+			throw new TypeError('Cannot construct Entity instances directly');
+		}
+		this._entity = entity;
+		this._sdkParentEntity = sdkParentEntity;
+	}
+	/**
+	 * Checks to see if the entity has className attached.
+	 * @param {*} className Class name you want to see if the entity has attached to it.
+	 */
+	hasClass(className) {
+		return this._entity && this._entity.hasClass(className);
+	}
+	/**
+	 * Get the action with an Action name that is attached to the entity
+	 * @param {*} actionName Action name to get the action attached to the entity.
+	 */
+	getActionByName(actionName) {
```

Comment:
https://github.com/BrightspaceHypermediaComponents/siren-sdk/blob/master/src/es6/Entity.js#L58-L63

The code duplication introduced here feels a bit off.  We should probably consolidate things into something logical?  Could be a simple utility? Worth thinking about